src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -1132,6 +1138,15 @@ public BoundExpression Convert(TypeSymbol type, BoundExpression arg, Conversion
                 return Null(type);
             }
 
+            if (conversion.Kind == ConversionKind.ExplicitNullable &&
+                arg.Type.IsNullableType() &&
+                arg.Type.GetNullableUnderlyingType().Equals(type, TypeCompareKind.IgnoreDynamicAndTupleNames))
+            {
+                // A conversion to unbox a nullable value is produced when binding a pattern-matching
+                // operation from an operand of type T? to a pattern of type T.
+                return this.Call(arg, this.SpecialMethod(CodeAnalysis.SpecialMember.System_Nullable_T_get_Value).AsMember((NamedTypeSymbol)arg.Type));
```

Comment:
I'm not confident that can occur, as it is likely to be found missing at a higher level in the caller. However, I'll add tests (`MissingNullable_03`) that *would* fail here if not for the fact that it is handled earlier.